lib/datadog/kit/identity.rb

```diff
@@ -56,9 +57,17 @@ def self.set_user(trace, id:, email: nil, name: nil, session_id: nil, role: nil,
         others.each do |k, v|
           trace.set_tag("usr.#{k}", v) unless v.nil?
         end
+
+        if ::Datadog::AppSec.settings.enabled
+          _, appsec_response = ::Datadog::AppSec::Instrumentation.gateway.push('identity.set_user', id)
+          if appsec_response && appsec_response.any? { |action, _event| action == :block }
+            throw(AppSec::Ext::REQUEST_INTERRUPT, [nil, appsec_response])
```

Comment:
I think we should create a return object rather than an array, which is the rack expected interface

Reply:
Yeah, also by being an object, it would have a type name, thus make it easier to track the expectations and usage from the sender to the receiver.

I feel this is out of scope for this PR though.